trio/_util.py

```diff
@@ -14,10 +14,23 @@
 import trio
 
 CallT = t.TypeVar("CallT", bound=t.Callable[..., t.Any])
+T = t.TypeVar("T")
+RetT = t.TypeVar("RetT")
+
+if t.TYPE_CHECKING:
+    from typing_extensions import ParamSpec, Self
+
+    ArgsT = ParamSpec("ArgsT")
+
+
+if t.TYPE_CHECKING:
+
+    def signal_raise(signum: int) -> None:
+        ...  # Don't check the implementation below, pthread_kill does not exist on Windows.
 
 
 # Equivalent to the C function raise(), which Python doesn't wrap
-if os.name == "nt":
+elif os.name == "nt":
```

Comment:
unrelated to PR, but I've mostly seen this as `sys.platform == "win32"` in other files. Any reason to prefer one over the other?

Reply:
Hmm, looking at the docs, `os.name` is more coarse, and would just give `posix` I think for Linux, MacOS, etc. But actually [type checkers](https://typing.readthedocs.io/en/latest/source/stubs.html#version-and-platform-checks) are only required to support `platform`.

Reply:
`git grep` gives `os.name` 15 times across 11 files, `sys.platform` 57 times across 29 files. Sounds like something for a dedicated PR

Reply:
Looking at SO, seems like `os.name`s coarseness has it's advantages when we don't care about mac/linux distinction if we want users to be able to attempt running `trio` on weird:tm: systems, as `sys.platform` can return values outside of win32/linux/darwin.